### PR TITLE
chore(test): async mongoose hooks error

### LINKED ISSUE #88: POST does not return error reported by asynchronous pre save
When a Mongoose model uses async pre save middleware and reports and error with done, the body of the POST response does not include the error.  So for example:
```javascript
schema.pre('save', true, function(err, model, next, done) {
    next();
    asyncDonewithError(done, 'ThisIsMyErrorMessage');
});
```
I can get a Mongoose unit test to return the error message, but when I invoke the REST interface, I only get a status code of 400 (Bad Request) and an empty body.

However, when I trigger simple schema validation (either custom validation function or violating unique constraints) the REST interface returns status code of 400 with the mongoose error detail in the body.

Is there a way to get Mongoose middleware errors to propagate like normal validation errors?

### PR DESCRIPTION
**Pull request by Zertz**
Adds tests for async mongoose pre save hooks.

Status code correctly returns an error, but the error itself is not being passed along. Investigating.

So it turns out there is a significant issue in how non-Mongoose errors are handled internally. Playing around with some solutions, but I think this will be a (small) breaking change.

Closes #88

### CODE REVIEW

File: src/errorHandler.js [hunk 1]
```diff
@@ -1,10 +1,12 @@
 module.exports = function (options) {
   return function (req, res, next) {
     return function (err) {
+      req.erm = req.erm || {}
+
       if (req.params.id && err.path === options.idProperty && err.name === 'CastError') {
-        err.statusCode = 404
+        err.statusCode = req.erm.statusCode = 404
       } else {
-        err.statusCode = 400
```
Comment (Zertz) [inline src/errorHandler.js]: This is where the bug was coming from.

**Comment by Zertz** [comment]
@florianholzapfel The fix is a breaking change, it has to do with the `Error` object not being serialized correctly when trying to add a property (`err.statusCode` in this code). I think for the sake of consistency, we should remove all use of `err.statusCode` and set `req.erm.statusCode` instead since this is where it is for successful operations.

The one failing test is an interesting edge case. The operation on the database succeeded, but the post middleware yields an error. Not exactly sure how this should be handled. Right now it's confusing because the response is `201` but the body says `400`.

**Comment by florianholzapfel** [comment]
@zertz you are right, we should use ```req.erm.statusCode``` instead of ```err.statusCode```.

**Comment by Zertz** [comment]
Ugh, something's causing a couple tests to fail in Node 0.10 :crying_cat_face:
